### PR TITLE
EE-401: removing session repository

### PR DESCRIPTION
**Pull request by garethgooch**

### CODE REVIEW

File: test/controllers/authenticated/AuthRedirectControllerSpec.scala [hunk 1]
```diff
@@ -36,31 +35,29 @@ import scala.concurrent.Future
...
 class AuthRedirectControllerSpec extends SpecBase with ScalaFutures with IntegrationPatience with MockitoSugar {
 
+  private val mockSessionRepository = mock[SessionRepository]
+
+  when(mockSessionRepository.set(any(), any())) thenReturn Future.successful(true)
+
   "AuthRedirectController" must {
 
-    "redirect to TaxYearSelection on success" in {
+    "redirect to TaxYearSelection on success and test integration with session repo works as expected" in {
```
Comment (Pascoe121) [inline test/controllers/authenticated/AuthRedirectControllerSpec.scala]: actually needs to verify the set worked.

File: test/controllers/construction/JoinerCarpenterControllerSpec.scala [hunk 2]
```diff
@@ -33,12 +36,17 @@ import play.api.test.Helpers._
 import repositories.SessionRepository
 import views.html.construction.JoinerCarpenterView
 
-class JoinerCarpenterControllerSpec extends SpecBase with ScalaFutures with IntegrationPatience with OptionValues {
+import scala.concurrent.Future
+
+class JoinerCarpenterControllerSpec extends SpecBase with ScalaFutures with IntegrationPatience with OptionValues with MockitoSugar with BeforeAndAfterEach {
```
Comment (Pascoe121) [inline test/controllers/construction/JoinerCarpenterControllerSpec.scala]: added beforeAndAfterEach but doesn't use it

File: test/controllers/construction/StoneMasonControllerSpec.scala [hunk 3]
```diff
@@ -33,12 +36,17 @@ import play.api.test.Helpers._
 import repositories.SessionRepository
 import views.html.construction.StoneMasonView
 
-class StoneMasonControllerSpec extends SpecBase with ScalaFutures with IntegrationPatience with OptionValues {
+import scala.concurrent.Future
+
+class StoneMasonControllerSpec extends SpecBase with ScalaFutures with IntegrationPatience with OptionValues with MockitoSugar with BeforeAndAfterEach {
```
Comment (Pascoe121) [inline test/controllers/construction/StoneMasonControllerSpec.scala]: added beforeAndAfterEach but doesn't use it

File: test/controllers/electrical/ElectricalControllerSpec.scala [hunk 4]
```diff
@@ -34,12 +37,17 @@ import play.api.test.Helpers._
 import repositories.SessionRepository
 import views.html.electrical.ElectricalView
 
-class ElectricalControllerSpec extends SpecBase with ScalaFutures with IntegrationPatience with OptionValues {
+import scala.concurrent.Future
+
+class ElectricalControllerSpec extends SpecBase with ScalaFutures with IntegrationPatience with OptionValues with MockitoSugar with BeforeAndAfterEach {
```
Comment (Pascoe121) [inline test/controllers/electrical/ElectricalControllerSpec.scala]: added beforeAndAfterEach but doesn't use it

File: test/controllers/healthcare/AmbulanceStaffControllerSpec.scala [hunk 5]
```diff
@@ -34,12 +37,17 @@ import play.api.test.Helpers._
 import repositories.SessionRepository
 import views.html.healthcare.AmbulanceStaffView
 
-class AmbulanceStaffControllerSpec extends SpecBase with ScalaFutures with IntegrationPatience with OptionValues {
+import scala.concurrent.Future
+
+class AmbulanceStaffControllerSpec extends SpecBase with ScalaFutures with IntegrationPatience with OptionValues with MockitoSugar with BeforeAndAfterEach {
```
Comment (Pascoe121) [inline test/controllers/healthcare/AmbulanceStaffControllerSpec.scala]: added beforeAndAfterEach but doesn't use it

File: test/controllers/healthcare/AmbulanceStaffControllerSpec.scala [hunk 6]
```diff
@@ -22,8 +22,11 @@ import controllers.actions.UnAuthed
 import forms.healthcare.AmbulanceStaffFormProvider
 import models.{NormalMode, UserAnswers}
 import navigation.{FakeNavigator, Navigator}
-import org.scalatest.OptionValues
+import org.mockito.Matchers.any
+import org.mockito.Mockito.{reset, times, verify, when}
```
Comment (Pascoe121) [inline test/controllers/healthcare/AmbulanceStaffControllerSpec.scala]: dosn't look like reset is used either. Optimise imports.

File: test/controllers/healthcare/HealthcareList1ControllerSpec.scala [hunk 7]
```diff
@@ -33,12 +36,17 @@ import play.api.test.Helpers._
 import repositories.SessionRepository
 import views.html.healthcare.HealthcareList1View
 
-class HealthcareList1ControllerSpec extends SpecBase with ScalaFutures with OptionValues with IntegrationPatience {
+import scala.concurrent.Future
+
+class HealthcareList1ControllerSpec extends SpecBase with ScalaFutures with OptionValues with IntegrationPatience with MockitoSugar with BeforeAndAfterEach {
```
Comment (Pascoe121) [inline test/controllers/healthcare/HealthcareList1ControllerSpec.scala]: added beforeAndAfterEach but doesn't use it

File: test/controllers/healthcare/HealthcareList1ControllerSpec.scala [hunk 8]
```diff
@@ -22,8 +22,11 @@ import controllers.actions.UnAuthed
 import forms.healthcare.HealthcareList1FormProvider
 import models.{NormalMode, UserAnswers}
 import navigation.{FakeNavigator, Navigator}
-import org.scalatest.OptionValues
+import org.mockito.Matchers.any
+import org.mockito.Mockito.{reset, times, verify, when}
```
Comment (Pascoe121) [inline test/controllers/healthcare/HealthcareList1ControllerSpec.scala]: Can't see reset use

File: test/controllers/healthcare/HealthcareList2ControllerSpec.scala [hunk 9]
```diff
@@ -33,13 +36,21 @@ import play.api.test.Helpers._
 import repositories.SessionRepository
 import views.html.healthcare.HealthcareList2View
 
-class HealthcareList2ControllerSpec extends SpecBase with ScalaFutures with IntegrationPatience with OptionValues {
+import scala.concurrent.Future
+
+class HealthcareList2ControllerSpec extends SpecBase with ScalaFutures with IntegrationPatience with OptionValues with MockitoSugar with BeforeAndAfterEach {
```
Comment (Pascoe121) [inline test/controllers/healthcare/HealthcareList2ControllerSpec.scala]: added beforeAndAfterEach but doesn't use it

File: test/controllers/healthcare/HealthcareList2ControllerSpec.scala [hunk 10]
```diff
@@ -22,8 +22,11 @@ import controllers.actions.UnAuthed
 import forms.HealthcareList2FormProvider
 import models.{NormalMode, UserAnswers}
 import navigation.{FakeNavigator, Navigator}
-import org.scalatest.OptionValues
+import org.mockito.Matchers.any
+import org.mockito.Mockito.{reset, times, verify, when}
```
Comment (Pascoe121) [inline test/controllers/healthcare/HealthcareList2ControllerSpec.scala]: reset not used

File: test/controllers/manufacturing/AluminiumOccupationList1ControllerSpec.scala [hunk 11]
```diff
@@ -22,22 +22,36 @@ import controllers.actions.UnAuthed
...
-  val formProvider = new AluminiumOccupationList1FormProvider()
-  val form = formProvider()
+  private val formProvider = new AluminiumOccupationList1FormProvider()
+  private val form = formProvider()
+  private val userAnswers = emptyUserAnswers
+  private val mockSessionRepository = mock[SessionRepository]
+
+  when(mockSessionRepository.set(any(), any())) thenReturn Future.successful(true)
+
+  def sessionApplication: Application = applicationBuilder(userAnswers = Some(userAnswers))
```
Comment (Pascoe121) [inline test/controllers/manufacturing/AluminiumOccupationList1ControllerSpec.scala]: Why did we suddenly change pattern. All other implementation follows the same process, I don't see why this one should be different. Either do this for all or keep this inline with all other test files.

File: test/controllers/FirstIndustryOptionsControllerSpec.scala [hunk 12]
```diff
@@ -147,22 +160,37 @@ class FirstIndustryOptionsControllerSpec extends SpecBase with ScalaFutures with
...
+      case _ => userAnswers
+        .set(FirstIndustryOptionsPage, trade).success.value
+    }
 
-    val application: Application = applicationBuilder(userAnswers = Some(emptyUserAnswers))
-      .build()
+    s"save correct amount to ClaimAmount when '$trade' is selected" in {
 
-    val sessionRepository = application.injector.instanceOf[SessionRepository]
+      reset(mockSessionRepository)
```
[the remark this answers is not in the document]
Comment (garethgooch) [reply test/controllers/FirstIndustryOptionsControllerSpec.scala]: No, this is hangover from trying to resolve the issues with the test, removed the unnecessary line

File: test/controllers/healthcare/HealthcareList1ControllerSpec.scala (hunk 7 above)
Comment (garethgooch) [reply test/controllers/healthcare/HealthcareList1ControllerSpec.scala]: Removed from all unused tests

**Pull request merged**